internal/transformer/classes/generators/dockerfilegenerators/PHPdockerfilegenerator.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	phpExt      = ".php"
+	virtualHost = "VirtualHost"
+)
+
+// PHPDockerfileGenerator implements the Transformer interface
+type PHPDockerfileGenerator struct {
+	Config transformertypes.Transformer
+	Env    *environment.Environment
+}
+
+// PhpTemplateConfig implements Php config interface
+type PhpTemplateConfig struct {
+	Ports    []int32
+	ConfFile string
+}
+
+// Init Initializes the transformer
+func (t *PHPDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *PHPDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// BaseDirectoryDetect runs detect in base directory
+func (t *PHPDockerfileGenerator) BaseDirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	return nil, nil, nil
+}
+
+// DetectConf extracts the port information from .conf file
+func DetectConf(dir string) (int32, string, error) {
+	var port int32
+	var confFileName string
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory : %s", err)
+		return port, confFileName, err
+	}
+	for _, de := range dirEntries {
+		ext := filepath.Ext(de.Name())
+		if ext != ".conf" {
+			continue
+		}
+		confFileName := de.Name()
+		confFilePath := filepath.Join(dir, de.Name())
+		confFile, err := os.Open(confFilePath)
+		if err != nil {
+			logrus.Errorf("Could not open the conf file: %s", err)
+			return port, confFileName, err
+		}
+		root, err := gopache.Parse(confFile)
+		if err != nil {
+			logrus.Errorf("Error while parsing configuration file : %s", err)
+			return port, confFileName, err
+		}
+		match, err := root.FindOne(virtualHost)
+		if err != nil {
+			logrus.Errorf("Could not find the VirtualHost in conf file: %s", err)
+			return port, confFileName, err
+		}
+		tokens := strings.Split(match.Content, ":")
+		if len(tokens) > 1 {
+			port, err := strconv.ParseInt(tokens[1], 10, 32)
+			if err != nil {
+				logrus.Errorf("Error while converting the entered port from string to int : %s", err)
+				return int32(port), confFileName, err
+			}
+			return int32(port), confFileName, nil
+		}
+		defer confFile.Close()
+	}
+	return port, confFileName, err
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PHPDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory : %s", err)
+		return nil, nil, err
+	}
+	for _, de := range dirEntries {
+		ext := filepath.Ext(de.Name())
```

Comment:
Why not being assignment and check into one line, if we are not doing anything else with ext. Also it could be a negative check and continue to avoid indentation

Reply:
Yes sure, I will make that change.